patches/common/chromium/printing.patch

```diff
@@ -0,0 +1,1010 @@
++      base::Callback<void(v8::Local<v8::Value>, v8::Local<v8::Value>)>;
++
+   // content::WebContentsObserver implementation.
+   bool OnMessageReceived(const IPC::Message& message,
+                          content::RenderFrameHost* render_frame_host) override;
+
++  void PrintToPDF(const base::DictionaryValue& options,
++                  const PrintToPDFCallback& callback);
++
+  private:
+   explicit PrintPreviewMessageHandler(content::WebContents* web_contents);
+   friend class content::WebContentsUserData<PrintPreviewMessageHandler>;
+
++  typedef std::map<int, PrintToPDFCallback> PrintToPDFCallbackMap;
++
++#if 0
+   // Gets the print preview dialog associated with the WebContents being
+   // observed.
+   content::WebContents* GetPrintPreviewDialog();
+@@ -72,12 +82,14 @@ class PrintPreviewMessageHandler
+   void OnDidPreviewPage(content::RenderFrameHost* render_frame_host,
+                         const PrintHostMsg_DidPreviewPage_Params& params,
+                         const PrintHostMsg_PreviewIds& ids);
++#endif
+   void OnMetafileReadyForPrinting(
+       content::RenderFrameHost* render_frame_host,
+       const PrintHostMsg_DidPreviewDocument_Params& params,
+       const PrintHostMsg_PreviewIds& ids);
+   void OnPrintPreviewFailed(int document_cookie,
+                             const PrintHostMsg_PreviewIds& ids);
++#if 0
+   void OnPrintPreviewCancelled(int document_cookie,
+                                const PrintHostMsg_PreviewIds& ids);
+   void OnInvalidPrinterSettings(int document_cookie,
+@@ -104,6 +116,10 @@ class PrintPreviewMessageHandler
+                                   const PrintHostMsg_PreviewIds& ids,
+                                   mojom::PdfCompositor::Status status,
+                                   base::ReadOnlySharedMemoryRegion region);
++#endif
++  void RunPrintToPDFCallback(int request_id, uint32_t data_size, char* data);
++
++  PrintToPDFCallbackMap print_to_pdf_callback_map_;
+
+   base::WeakPtrFactory<PrintPreviewMessageHandler> weak_ptr_factory_;
+
+diff --git a/chrome/browser/printing/print_view_manager_base.cc b/chrome/browser/printing/print_view_manager_base.cc
+index 9b0c15b3c501..95be20eed741 100644
+--- a/chrome/browser/printing/print_view_manager_base.cc
++++ b/chrome/browser/printing/print_view_manager_base.cc
+@@ -30,7 +30,7 @@
+ #include "chrome/browser/ui/simple_message_box.h"
+ #include "chrome/browser/ui/webui/print_preview/printer_handler.h"
+ #include "chrome/common/pref_names.h"
+-#include "chrome/grit/generated_resources.h"
++#include "electron/grit/electron_resources.h"
+ #include "components/prefs/pref_service.h"
+ #include "components/printing/browser/print_composite_client.h"
+ #include "components/printing/browser/print_manager_utils.h"
+@@ -67,6 +67,8 @@ using PrintSettingsCallback =
+     base::OnceCallback<void(scoped_refptr<PrinterQuery>)>;
+
+ void ShowWarningMessageBox(const base::string16& message) {
++  LOG(ERROR) << "Invalid printer settings";
```

Comment:
perhaps also include the message?

Reply:
still think this should have the message in it as well, e.g. `LOG(ERROR) << "Invalid printer settings: " << message;`